NugetLightRequest.cs

```diff
@@ -1561,11 +1563,34 @@ private string ConfigurationFileLocation
                         && (new SemanticVersion(minimumVersion) == new SemanticVersion(maximumVersion))
                         && minInclusive && maxInclusive);
 
-                // if required version not specified then don't use unlisted version
-                // unlisted version is the one that has published year as 1900
+                // find out if all versions of a package are unlisted
+                // unlisted versions will have a published year as 1900 or earlier
+                var listedPkgs = pkgs.Where(p => (!p.Published.HasValue || p.Published.Value.Year > 1900));
+                var allVersionsUnlisted= (listedPkgs.IsNullOrEmpty()) ? true : false;
+
                 if (!exactVersionRequired)
                 {
-                    pkgs = pkgs.Where(pkg => (!pkg.Published.HasValue || pkg.Published.Value.Year > 1900));
+                    // if at least one version is listed, return the set of listed packages.
+                    // if all verisons are unlisted and package is not a dependency, don't return any packages
+                    // if all versions are unlisted and it is a dependency, just return all the packages
+                    if (!allVersionsUnlisted)
+                    {
+                        pkgs = listedPkgs;
+                    }
+                    else if (allVersionsUnlisted && !isDependency)
+                    {
+                        pkgs = null;
+                    }
+
+                    if (minimumVersion != null && !minInclusive)
+                    {
+                        pkgs = pkgs.Where(p => (p.Version != (new SemanticVersion(minimumVersion))));
+                    }
+
+                    if (maximumVersion != null && !maxInclusive)
+                    {
+                        pkgs = pkgs.Where(p => (p.Version != (new SemanticVersion(maximumVersion))));
+                    }
```

Comment:
Could you please explain the need for this?
It looks like below code at line#1637 is handling the same.

                if (findResult.VersionPostFilterRequired)
                {
                    pkgs = PackageFilterUtility.FilterOnVersion(pkgs, requiredVersion, minimumVersion, maximumVersion, minInclusive, maxInclusive);
                }